### ISSUE TITLE
Events

### ISSUE THREAD

**Issue by richardofsussex** [national-gallery/NG-CIIM#19]
I've had a go at Events today, mapping them to the Linked Art framework. See:

http://richardofsussex.me.uk/ng/output/event-7.json
http://richardofsussex.me.uk/ng/output/event-22.json
http://richardofsussex.me.uk/ng/output/event-25.json

I have made no attempt to represent the recurring nature of the recurring event (Christmas day), but otherwise the results should be worth checking.

**Comment by jpadfield** [comment]
I do think the empty classified_as and referred_to_by could be a problem - if we do not have data can we just omit them? Do we want to consider the notion of stub nodes  ... basically place holders indicating when we know we do not have data ... might make it clearer and easier to identify gaps ..

The use of "inverted terms" here does not really make sense, as the terms have not been inverted ...

Do we have any additional information relating to the difference of the two name? Only one is marked "_label"  would we want long/short or alternative, ..... Not needed just wondering.

The Timespan issue could be complicated and again an issue relating to what we do when we have no data - as noted before I did do work a long time ago to parse all of our dates and identify events and start and finishes and overlapping decades and centuries. we could reuse some of this if it helps ....

In general I think we need to be consistent with how we format dates for all entities we express - LA uses the ''1763-04-01T00:00:00Z' type format and uses the other terms as well "end_of_the_beginning" it might be good to do this ... we just need a consistent set of rules to determine a range from a given date description (something like https://research.ng-london.org.uk/wiki/index.php/National_Gallery_Display_Date_Descriptors)

If we really wanted to us "dates" when we do have any we could just use broad ones ....

Again this comes to how we document things - if dates are know you get these data points, if data points are not included it means dates are unknown.

Do we have a list of all of the dates that have values but do not relate to actual numbers ....

**Comment by richardofsussex** [comment]
"inverted terms" made sense for personal names (agent); not so here as you say. Here are its sibling concepts:
![image](https://github.com/national-gallery/NG-CIIM/assets/2197102/df2d166a-38db-46f8-ad5d-e2adf63abbfd)
Is any of them a better match for your concept 'sort name'?

**Comment by RGShepherd** [comment]
6. None of those; but http://vocab.getty.edu/aat/300451544 seems ideal!

**Comment by richardofsussex** [comment]
Agreed: done.

**Comment by RGShepherd** [comment]
1. to do: please remove empty arrays.
2. resolved.
3. to do: rather than `classified_by`, let's use our agreed string + uncool URL for now
4. resolved.
5. = 1
6. resolved.
7. = 7
8. data here is pretty limited; I'll look at all the values we have. The one problem - as mentioned earlier - is events that are cyclical, which are not amenable to encoding as ISO dates. Then we can look at how best to convert values / strings to CRM-like dates.
9. = 8
10. = 8
11. = 8
12. = 8

**Comment by RGShepherd** [comment]
Re. **8**, here are all our current event date values:

| EventType | DateText | DateTimeBegin | DateTimeEnd | Count | 
| --- | --- | --- | --- | --- | 
| festival | NULL | NULL | NULL | 1 | 
| historic event | NULL | NULL | NULL | 2 | 
| festival | 25 December | NULL | NULL | 1 | 
| festival | 6 January | NULL | NULL | 1 | 
| historic event | 9 June 1156 | 1156-06-09 | 1156-06-09 | 1 | 
| historic event | 1 June 1432 | 1432-06-01 | 1432-06-01 | 1 | 
| historic event | 1517 | 1517 | NULL | 1 | 
| historic event | 12 February 1554 | 1554-02-12 | 1554-02-12 | 1 | 
| historic event | 7 October 1571 | 1571-10-07 | 1571-10-07 | 1 | 
| historic event | 22 October 1642 - 3 September 1651 | 1642-10-22 | 1651-09-03 | 1 | 
| historic event | 15 May 1648 | 1648-05-16 | 1648-05-16 | 1 | 
| historic event | 13 June 1665 | 1665-06-13 | 1665-06-13 | 1 | 
| historic event | 19 April 1691 | 1691-04-19 | 1691-04-19 | 1 | 
| historic event | 17 May 1756 - 15 February 1763 | 1756-05-17 | 1763-02-15 | 1 | 
| historic event | 20 April 1792 - 20 November 1815 | 1792-04-20 | 1815-11-20 | 1 | 
| historic event | 20 September 1792 | 1792-09-20 | 1792-09-20 | 1 | 
| historic event | 6 November 1792 | 1792-11-06 | 1792-11-06 | 1 | 
| historic event | 16 May 1795 | 1795-05-16 | 1795-05-16 | 1 | 
| historic event | 21 October 1805 | 1805-10-21 | 1805-10-21 | 1 | 
| historic event | 30 October 1813 | 1813-10-30 | 1813-10-30 | 1 | 
| historic event | 11 February 1814 | 1814-02-11 | 1814-02-11 | 1 | 
| historic event | 19 June 1867 | 1867-06-19 | 1867-06-19 | 1 | 
| historic event | 18 March 1871 - 28 May 1871 | 1871-03-18 | 1871-05-28 | 1 | 
| historic event | 1 April 1877 - 30 April 1877 | 1877-04-01 | 1877-04-30 | 1 | 
| (not assigned) | 15 THURSDAY | 2005-12-15 | 2005-12-15 | 1 |

So by-and-large, I'd say not much date processing needed. Our only real problems are:
- cyclical events ('festivals'), simply not amenable to ISO date formatting; 
- and (currently) `event-25`, 'The Reformation', which traditionally began with Luther nailing his theses to the door in 1517, but ended ... when? (Discuss.) https://linked.art/model/base/#time-span-details is not helpful as to events with no clearly-defined ending.

**Comment by richardofsussex** [comment]
I took it from this list that conversion from the DateText values to the DateTimeBegin and DateTimeEnd ones was required, and have done this in the XSLT. However, examining the data I see that all the DateTime values are already recorded in date.from and date.to. Anyway, the code is there should it be needed, and at least it now adds a suitable 'start of day' and 'end of day' Time qualifier to yyyy-mm-dd dates.

**Comment by RGShepherd** [comment]
Yes, I think safest to stick with conversion of the YYYY[-MM-DD] values in the from and to fields, rather than the free text - likely to be less fragile.

The pedant in me (and I think ISO-8601) thinks we should drop the timezone 'Z' as the times should be assumed to be  local / predate the adoption of timezones.

Any thoughts on how to handle timespans without defined beginnings / ends?

**Comment by richardofsussex** [comment]
Timezone dropped (and support for yyyy-mm added - missed that case).

I think what we have now is logical. If only the start of the timespan is known, just record that fact. Timespans don't need to be complete.

**Comment by RGShepherd** [comment]
Is there an example I can check? Timezone still appearing in e./g. event-7:

```
"timespan": {
  "type": "TimeSpan",
  "begin_of_the_begin": "1792-11-06T00:00:00Z",
  "end_of_the_end": "1792-11-06T23:59:59Z",
  "_label": "6 November 1792"
},
```

**Comment by richardofsussex** [comment]
Try now - I failed to actually re-process the events! 7 and 25 should be OK now.

**Comment by RGShepherd** [comment]
Thanks! please could you also process event-9, so I can double-check an entry with different start and end dates? Looking good so far ...

**Comment by richardofsussex** [comment]
Done: http://richardofsussex.me.uk/ng/ciim7-output/event-9.json
